CSS module project: multi-column layout

estelle opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

<!-- ✍️ Summarize your changes in one or two sentences -->

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

estelle committed:
fix conflicts

estelle committed:
fix conflicts

estelle committed:
@#$% merge conflicts

estelle committed:
@#$% merge conflicts

estelle committed:
update content now that merging is resolved.

estelle committed:
update content now that merging is resolved.

ChrisCoastal committed:
fix typo scrollbar_role (#27051)

estelle committed:
see also

estelle committed:
see also

estelle committed:
see also

estelle committed:
see also

estelle committed:
intro paragraph

estelle committed:
intro paragraph

estelle committed:
add a see also

estelle committed:
add a see also

aredridel committed:
Clarify meaning of "object property" when iterating (#27055)

* Clarify meaning of "object property" when iterating

* Propagate change

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

iugo committed:
feat: add tips for string length (#27045)

Olatade committed:
Update typo in "unregistering a period sync" subtopic (PWAs)  (#27012)

Update index.md

mfuji09 committed:
fix a typo in CSSStyleSheet/ownerRule (#27009)

fix a typo

rschristian committed:
Fix broken link references (#27020)

* fix: Reference from spanindex to rowindex

* Update index.md

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

ya-sh-vardh-an committed:
Change "<option>s" to "<option> elements" (#27003)

* Fix typo

Replace <option>s with <option>

* Update index.md

Add elements in place of s

Commit:
WebAssembly | Memory - use IEC standard measurement (#27016)

WebAssembly | Memory - Use IEC standard measurement

Due to an oversight, a measurement was not denoted with IEC.

wbamberg committed:
PWA How-to: trigger installation from a PWA (#26842)

* PWA How-to: trigger installation from a PWA

* copy edit

* Apply suggestions from code review

Co-authored-by: Patrick Brosset <patrickbrosset@gmail.com>

* Review comments

* Update files/en-us/web/progressive_web_apps/how_to/trigger_install_prompt/index.md

Co-authored-by: Patrick Brosset <patrickbrosset@gmail.com>

---------

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>
Co-authored-by: Patrick Brosset <patrickbrosset@gmail.com>

yin1999 committed:
chore: remove tailing slash of MDN docs' urls (#27036)

* chore: remove tailing slash of MDN docs' urls

* fix broken links in those pages

wbamberg committed:
Fix issue 26841 (#26952)

* Fix 26841

* Fix description

Josh-Cena committed:
Consistently capitalize Wasm (#27062)

geryogam committed:
Use more similar examples for private static fields and methods (#26055)

* Use more similar descriptions between private static fields and methods

* Remove trailing spaces

* Remove accesses to private static members through this in examples (only keep them in counter-examples)

* Update index.md

* Uniformise the descriptions of private fields and private methods

* Some fixes

* Update index.md

* Clarify mentions of the prototype property

* Little capitalization

---------

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>
Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

vch-7 committed:
Fix iframe session history description (#27022)

* Fixed iframe session history docs

* caps

* dup

kalu06 committed:
Update index.md (#27064)

dependabot[bot] committed:
Bump markdownlint-rule-search-replace from 1.0.10 to 1.1.0 (#27072)

OnkarRuikar committed:
Synchronize with BCD v5.2.60 (#27057)

Co-authored-by: OnkarRuikar <OnkarRuikar@users.noreply.github.com>

SphinxKnight committed:
Improve example for `margin-left` (#27067)

* improve example

* fix linting errors

* Update files/en-us/web/css/margin-left/index.md

---------

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>

yarusome committed:
Update `<easing-function>` (#27071)

* Update `<easing-function>`

* Fix links

* Fine-tune

dumb-programmer committed:
Update index.md (#27080)

OscarNOW committed:
Fix wrong column header in image types (#27077)

There is a column with the header "Description" but inside the column are the supported browsers, so I'm proposing to change the header to "Supported browsers"

geryogam committed:
Fix word order for static/instance and public/private (#27081)

* Fix word order

* Update index.md

geryogam committed:
Fix case in single-sentence lists (#27082)

jimmonte committed:
clarified distinction between call and apply (#27013)

* clarified distinction between call and apply

Added detail that apply can accept an array-like object and reversed the order of the examples to match the previous order in the sentence.

const numbers = [5, 6, 2, 3, 7];

const max = Math.max.apply(null, numbers);

console.log(max);
// Expected output: 7

const min = Math.min.apply(null, numbers);

console.log(min);
// Expected output: 2

const num2 = {
  length: 5,
  0: 5,
  1: 6,
  2: 2,
  3: 3,
  4: 7,
};

const min2 = Math.min.apply(null, num2);
const max2 = Math.max.apply(null, num2);

console.log(max2);
console.log(min2);

* Minor revision to text of note

* Propagate change

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

Morteza-Porabdolahi committed:
Fix typo (#27088)

Update index.md

Fix invalid object indexing with strings

Josh-Cena committed:
Avoid using non-standard date string format, better date string description (#27005)

* Avoid using non-standard date string format, better date string description

* Fixes

* Apply suggestions from code review

Co-authored-by: Florian Scholz <fs@florianscholz.com>

---------

Co-authored-by: Florian Scholz <fs@florianscholz.com>

mfuji09 committed:
Remove an unnecessary section (#27101)

viceodev committed:
Update index.md (#27099)

viceodev committed:
Fix grammar error in WebExtensions cookies.set docs (#27095)

Update index.md

minimalsm committed:
Fix broken link on device orientation events WEB APIs docs  (#27094)

Fix broken link to web.dev blog post

xenomote committed:
Update link to color-mix documentation (#27103)

Josh-Cena committed:
Add note that :target does not work in shadow roots (#27089)

santiago-dev96 committed:
Add missing connector (#27086)

add missing connector

OnkarRuikar committed:
Update discord invite (#27068)

krave1986 committed:
Link to the \xAB notation (#27097)

* Link to the \xAB notation

* Fix lint-docs error

Fenced code blocks should be surrounded by blank lines [Context: "```"]

* Update index.md

bsmth committed:
docs(relnote): Fx114 - crossorigin attribute for image and feImage elements in SVGs (#27065)

* docs(relnote): Fx114 - crossorigin attribute for image and feImage elements in SVGs

* docs(relnote): Fx114 - crossorigin attribute for image and feImage elements in SVGs

HolgerJeromin committed:
Reference iframe in getRootNode() API (#27108)

wbamberg committed:
New PWA main page (#26736)

* New PWA main page

* Copy guides preamble

* Sync landing page and guides page

* Auto-generate interstitial pages

* Add some additional links

---------

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>

Zectbumo committed:
removed extraneous characters on element/template (#27118)

Elchi3 committed:
Remove docs for no longer relevant Proximity Events API (#27115)

Elchi3 committed:
WebGLRenderingContext has no commit method (#27114)

postmeback committed:
Update index.md (#26960)

Modified the iframe parameter with 120px.

Tunzeki committed:
Fix typo: remove "an" (#27122)

The explanation provided for `<linear-color-stop>` reads in part:

A color-stop's `<color>` value, followed by an one or two optional stop positions...

It should read:

A color-stop's `<color>` value, followed by one or two optional stop positions...

Rob--W committed:
DNR docs: replace /docs with /en-US/docs + fix missing WebExtAPIRef (#26818)

* DNR docs: fix mdn flaws

- Change /docs/ links to /en-US/docs/
- Add missing `{{WebExtAPIRef` in declarativenetrequest/index.md

* Update files/en-us/mozilla/add-ons/webextensions/api/declarativenetrequest/index.md

Co-authored-by: rebloor <git@sherpa.co.nz>

---------

Co-authored-by: Queen Vinyl Da.i'gyu-Kazotetsu <vinyldarkscratch@gmail.com>
Co-authored-by: rebloor <git@sherpa.co.nz>

chrisn committed:
Add missing space (#27135)

hamishwillee committed:
Introduction to the CSS basic box model - remove spec section (#27147)

Commit:
WebAssembly | Memory - add note about allocation failure (#27017)

* WebAssembly | Memory - add note about allocation failure

* memory/memory - add note about max initial.

* memory/memory - revert irrelevant change

* memory/memory - remove DOM/web page assumption

* memory/memory - rewrite exceptions section to be a list

* memory/grow - add note about SAB regarding detachment

* revert last commit - accident; wrong branch

* Update index.md

---------

Co-authored-by: Joshua Chen <sidachen2003@gmail.com>

caugner committed:
fix: update og:image (#27046)

* fix: update og:image

opengraph-logo.png has been replaced by mdn-social-share.png

* fix: use absolute og:image url

minimalsm committed:
Add a note on browser support for desktop orientation events (#27139)

* Update device orientation events

* Update files/en-us/web/api/device_orientation_events/index.md

bsmth committed:
docs(relnote): Fx114 - ARIA attribute reflection enabled on nightly, landmark roles bugfix (#27117)

* docs(relnote): Fx114 - ARIA attribute reflection enabled in nightly

* docs(relnote): Fx114 - ARIA landmark roles bugfix

rhashimoto committed:
Correct return value description for LockManager.request (#27130)

The documentation for LockManager.request() claimed that the return value was a Promise that would resolve to undefined, which is not correct unless the callback returns undefined.

See the [spec](https://w3c.github.io/web-locks/#api-lock-manager-request):

> The returned promise resolves (or rejects) with the result of the callback after the lock is released, or rejects if the request is aborted.

OnkarRuikar committed:
Synchronize with BCD v5.2.61 (#27123)

Co-authored-by: OnkarRuikar <OnkarRuikar@users.noreply.github.com>

estelle committed:
Glossary updates: alignment to application-context (#26998)

* Glossary updates: alignment to application-context

* Apply suggestions from code review

Co-authored-by: Brian Thomas Smith <brian@smith.berlin>

---------

Co-authored-by: Brian Thomas Smith <brian@smith.berlin>

hamishwillee committed:
RTCStatsReport - make this the center of statistics docs (#27028)

* Add RTCStatsReport - docs where this is the centre

* minor mods

* Move the WebRTC Statistics API doc to Analyzing WebRTC statistics and cross linking

* Group data - remove undocumented/invalid WebRTC dictionaries and add Analyzing WebRTC Statistics

* RTCIceCandidatePairStats - remove RTCStats links (and other easy changes)

* Analyzing WebRTC Stats: remove RTCStats links

* RTCIceCandidateStats - remove RTCStats links (and other minimal changes)

* RTCIceCandidatePairStats - consistency fixes

* RTCInboundRtpStreamStats - remove RTCStats links (and other minimal changes)

* RTCOutboundRtpStreamStats - remove RTCStats links (and other minimal changes)

* RTCRemoteOutboundRtpStreamStats - remove RTCStats links (and other minimal changes)

* Remove remaining mentions of RTCStats

* Remove the RTCStats dictionary

* RTCInboundRtpStreamStats - add top level placeholders for stuff inherited from RTCRtpStreamStats

* RTCInboundRtpStreamStats - improve heading for RTCReceivedRtpStreamStats

* RTCIceCandidatePairStats - add the boilerplate-sh RTCStats id, timestamp, type

* Add boilerplate id, timestamp, type for existing dictionaries

* Apply easy suggestons from code review

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Update files/jsondata/GroupData.json

* Remove the analyzing doc - redundant

* Apply suggestions from code review

Co-authored-by: wbamberg <will@bootbonnet.ca>

* Remove invalid example

* Apply suggestions from code review

* RTCIceCandidateStats.candidateType - not optional

* Fix maplike methods to use preexisting peerconnection

* RTCStatsReport.size - improve text

* RTCStatsReport - section on the common instance variables

* Fix lint errors

* lint fix

---------

Co-authored-by: wbamberg <will@bootbonnet.ca>

hamishwillee committed:
Using the Web Speech API - remove old browser support info (#27146)

* Using the Web Speech API - remove old browser support info

* Run prettier

jespertheend committed:
Add page for Response.json() static method (#27154)

* Add page for Response.json() static method

* lint

tchajed committed:
Use decompress throughout DecompressionStream (#27158)

postmeback committed:
Update index.md (#27141)

Added the text given in the issue

yin1999 committed:
fix redirects of two links (#27132)

yin1999 committed:
chore: remove the note about activating frozen locales (#27073)

dipikabh commented on files/en-us/web/css/css_multicol_layout/index.md:
```diff
@@ -6,41 +6,37 @@ spec-urls: https://drafts.csswg.org/css-multicol/
-- {{cssxref("break-inside")}}
-- {{cssxref("orphans")}}
-- {{cssxref("widows")}}
+> **Note:** Setting container height and line length can pose challenges for people with visual or cognitive disabilities. WCAG Success Criterion state that, even when the text size is doubled, content should not need to be scrolled.
```
```suggestion
> **Note:** Bear in mind that setting container height and line length can pose challenges for people with visual or cognitive disabilities. WCAG Success Criterion states that even when the text size is doubled, content should not need to be scrolled.
```
could we add a link to the "WCAG Success Criterion"?

dipikabh commented on files/en-us/web/css/css_multicol_layout/index.md:
```diff
@@ -49,19 +45,31 @@ Multi-column layout is closely related to [paged media](/en-US/docs/Web/CSS/CSS_
-- [CSS flexible box layout](/en-US/docs/Web/CSS/CSS_flexible_box_layout) (CSS flexbox)
-- [CSS grid layout](/en-US/docs/Web/CSS/CSS_grid_layout)
+- [Learn: multiple-column layout](/en-US/docs/Learn/CSS/CSS_layout/Multiple-column_Layout)
+- [CSS paged media](/en-US/docs/Web/CSS/CSS_paged_media) module
+- [CSS fragmentation](/en-US/docs/Web/CSS/CSS_fragmentation) module
+- [CSS flexible box layout](/en-US/docs/Web/CSS/CSS_flexible_box_layout) module
+- [CSS grid layout](/en-US/docs/Web/CSS/CSS_grid_layout) module
```
```suggestion
- [CSS fragmentation](/en-US/docs/Web/CSS/CSS_fragmentation) module
- [CSS flexible box layout](/en-US/docs/Web/CSS/CSS_flexible_box_layout) module
- [CSS grid layout](/en-US/docs/Web/CSS/CSS_grid_layout) module
- [CSS paged media](/en-US/docs/Web/CSS/CSS_paged_media) module
```
moved 'paged media' to the end

dipikabh reviewed (changes requested):
Thanks @estelle. I've added a few edit suggestions.

estelle committed:
Minor: added guide to see also (#27054)

joshuakb2 committed:
Correct typo (#27159)

Refer to the correct operator at the top of the description.

captainbrosset committed:
PWA shortcuts how-to page (#27119)

* PWA shortcuts how-to page

* add new page to shortcut

* Apply suggestions from code review

Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>
Co-authored-by: wbamberg <will@bootbonnet.ca>

* Added link to main PWA page

---------

Co-authored-by: Estelle Weyl <estelle@weyl.org>
Co-authored-by: Estelle Weyl <estelle@openwebdocs.org>
Co-authored-by: wbamberg <will@bootbonnet.ca>

MrOrz committed:
Update Array.prototype.with() RangeError condition (#27163)

This is a trivial fix on the condition on when `Array.prototype.with()` throws `RangeError`.

wbamberg committed:
Add new how to on associating files with a PWA (#27155)

* Add new how to on associated files with a PWA

* Put API existence check back

* Compress images

* Update files/en-us/web/progressive_web_apps/how_to/associate_files_with_your_pwa/index.md

Co-authored-by: Patrick Brosset <patrickbrosset@gmail.com>

---------

Co-authored-by: Patrick Brosset <patrickbrosset@gmail.com>

dependabot[bot] committed:
Bump @mdn/yari from 2.23.1 to 2.24.0 (#27168)

whimboo committed:
WebDriver conformance changes for Firefox 114 (#27120)

* WebDriver conformance changes for Firefox 114

* Update files/en-us/mozilla/firefox/releases/114/index.md

Co-authored-by: Julian Descottes <julian.descottes@gmail.com>

---------

Co-authored-by: Julian Descottes <julian.descottes@gmail.com>

urffin committed:
Fix sample error in autoplay guide (#27157)

Update index.md

Fix error in code sample: wrong addEventListener and removeEventListener calls

estelle committed:
Apply suggestions from code review

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

estelle commented on files/en-us/web/css/css_multicol_layout/index.md:
```diff
@@ -6,41 +6,37 @@ spec-urls: https://drafts.csswg.org/css-multicol/
-## Relationship to fragmentation
-
-Multi-column layout is closely related to [paged media](/en-US/docs/Web/CSS/CSS_paged_media), in that each column box becomes a fragment, much like a printed page becomes a fragment of an overall document. Therefore, the properties now defined in the [CSS fragmentation](/en-US/docs/Web/CSS/CSS_fragmentation) specification are required to control how content breaks between columns.
+> **Note:** Multiple-column layout is closely related to [paged media](/en-US/docs/Web/CSS/CSS_paged_media), as each column box is a fragment, much like each printed page is a fragment of a document. Properties defined in [CSS fragmentation](/en-US/docs/Web/CSS/CSS_fragmentation) control how content breaks between columns.
```
Good catch!

estelle committed:
edits per PR comments

estelle commented:
Thanks for the review. I committed all the edits and addressed all the suggestions.

estelle committed:
merge issues

estelle committed:
merge issues
